src/init.cpp

```diff
@@ -1211,6 +1211,50 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     RegisterZMQRPCCommands(tableRPC);
 #endif
 
+    // Check port numbers
```

Comment:
In commit "fix: handle invalid rpcbind port earlier" (8892150c2d8ea9e09046e288cecd4c35472c0267)

Not important, but to make AppInitMain function shorter and make the bugfix commit fix more readable you could consider adding an earlier refactoring commit the moves these for loops into a `CheckHostPortOptions` function, and have AppinitMain call it like `if (!CheckHostPortOptions(args)) return false;`

Reply:
Thanks, good suggestion.  Makes things cleaner.  Added.
Was also thinking about updating `ParseUint16()` to `ToIntegral()`, but that could be left for a follow-up, since we would also want to ensure adequate tests for `-port` and `-rpcport` with that update.